### ISSUE TITLE
getAppsScriptFileName() outputs incorrect path separator on Windows in v1.6.3

### ISSUE THREAD

**Issue by sa3dany** [google/clasp#386]
## Expected Behavior
When using `clasp push` to upload files, files in sub-directories should map to subdir/file

## Actual Behavior
Files in sub-directories are mapped to subdir\file

## Specifications

  - Node version: 8.12.0
  - Version: 1.6.3
  - OS (Mac/Linux/Windows): Windows 10

It seems the transpiled code of `getAppsScriptFileName()` is the cause of this issue. Below I've included both the original TypeScript function and how it is transpiled:

```typescript
export function getAppsScriptFileName(rootDir: string, filePath: string) {
  const nameWithoutExt = filePath.slice(0, -path.extname(filePath).length);
  let fullFilePathNoExt = rootDir ? path.relative(rootDir, nameWithoutExt) : nameWithoutExt;

  // Replace OS specific path separator to common '/' char
  fullFilePathNoExt = fullFilePathNoExt.replace(/\\/g, '/');

  return fullFilePathNoExt;
}
```

```javascript
function getAppsScriptFileName(rootDir, filePath) {
    var nameWithoutExt = filePath.slice(0, -path.extname(filePath).length);
    // Replace OS specific path separator to common '/' char
    nameWithoutExt = nameWithoutExt.replace(/\\/g, '/');
    return rootDir ? path.relative(rootDir, nameWithoutExt) : nameWithoutExt;
}
```

In the transpiled version, a call to `path.relative` is made after the search and replace which returns a path with native path separators.

Since #48 was closed and the root cause appears to be different I've opted to open a new issue.

**Comment by takanakahiko** [comment]
Converting the latest source code in my environment will look like this.

```js
function getAppsScriptFileName(rootDir, filePath) {
    var nameWithoutExt = filePath.slice(0, -path.extname(filePath).length);
    var fullFilePathNoExt = rootDir ? path.relative(rootDir, nameWithoutExt) : nameWithoutExt;
    // Replace OS specific path separator to common '/' char
    fullFilePathNoExt = fullFilePathNoExt.replace(/\\/g, '/');
    return fullFilePathNoExt;
}
```

- Node version: 8.9.4
- OS (Mac/Linux/Windows): Mac

**Comment by sa3dany** [comment]
Just to make sure, I've used another Windows 10 PC with a fresh install of node (8.12.0) and clasp (1.6.3) and I get the same code as in the issue description.

```javascript
/**
 * Gets the name of the file for Apps Script.
 * Formats rootDir/appsscript.json to appsscript.json.
 * Preserves subdirectory names in rootDir
 * (rootDir/foo/Code.js becomes foo/Code.js)
 * @param {string} rootDir The directory to save the project files to.
 * @param {string} filePath Path of file that is part of the current project
 */
function getAppsScriptFileName(rootDir, filePath) {
    var nameWithoutExt = filePath.slice(0, -path.extname(filePath).length);
    // Replace OS specific path separator to common '/' char
    nameWithoutExt = nameWithoutExt.replace(/\\/g, '/');
    return rootDir ? path.relative(rootDir, nameWithoutExt) : nameWithoutExt;
}
```

**Comment by takanakahiko** [comment]
In v1.6.3, I confirm that the source code is wrong.

https://github.com/google/clasp/blob/16025391e99587922019a5d1375d6e77420a2b0a/src/files.ts#L199

However, the latest source code has been fixed.

https://github.com/google/clasp/blob/c1d1a6c5a679f47679d20b8dacf9cfad2d5834c9/src/files.ts#L205

This bug will cure on the next release.

**Comment by sa3dany** [comment]
Thanks. Do you know what caused the code to compile differently on v1.6.3? (just curious)

**Comment by takanakahiko** [comment]
This bug has been fixed in #351.
So I think it is okay to close this issue.
